dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala

```diff
@@ -226,15 +226,23 @@ case class BinaryOutcomeDLCClient(
   def createFundingTransactionSigs(): Future[FundingSignatures] = {
     val fundingTx = createUnsignedFundingTransaction
 
-    val sigFs = fundingUtxos.foldLeft(Vector.empty[Future[PartialSignature]]) {
+    val outPointAndSigFs = fundingUtxos.foldLeft(
+      Vector.empty[Future[(TransactionOutPoint, PartialSignature)]]) {
       case (vec, utxo) =>
-        val sigF = BitcoinSignerSingle.signSingle(utxo,
-                                                  fundingTx,
-                                                  isDummySignature = false)
+        val sigF = BitcoinSignerSingle
+          .signSingle(utxo, fundingTx, isDummySignature = false)
+          .map(sig => (utxo.outPoint, sig))
         vec :+ sigF
     }
+    val sigsF = Future.sequence(outPointAndSigFs)
 
-    Future.sequence(sigFs).map(FundingSignatures.apply)
+    val outPointAndSigsF = sigsF.map { sigs =>
+      sigs.map {
+        case (outPoint, _) =>
+          (outPoint, sigs.filter(_._1 == outPoint).map(_._2))
+      }.toMap
+    }
```

Comment:
I'm very confused by what is happening here and why this extra code is needed?

Reply:
It is turning a `Vector[(TransactionOutpoint, PartialSignature)]` into a `Map[TransactionOutpoint, Vector[PartialSignature]'